src/NuGet.Core/NuGet.Configuration/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,8 @@
+NuGet.Configuration.IPackageSourceProvider2
+NuGet.Configuration.IPackageSourceProvider2.SavePackageSources(System.Collections.Generic.IEnumerable<NuGet.Configuration.PackageSource> sources, NuGet.Configuration.PackageSourceUpdateOptions packageSourceUpdateOptions) -> void
+NuGet.Configuration.PackageSourceProvider.SavePackageSources(System.Collections.Generic.IEnumerable<NuGet.Configuration.PackageSource> sources, NuGet.Configuration.PackageSourceUpdateOptions sourceUpdateSettings) -> void
+NuGet.Configuration.PackageSourceUpdateOptions
+NuGet.Configuration.PackageSourceUpdateOptions.PackageSourceUpdateOptions(bool updateCredentials, bool updateEnabled) -> void
+NuGet.Configuration.PackageSourceUpdateOptions.UpdateCredentials.get -> bool
+NuGet.Configuration.PackageSourceUpdateOptions.UpdateEnabled.get -> bool
+static readonly NuGet.Configuration.PackageSourceUpdateOptions.Default -> NuGet.Configuration.PackageSourceUpdateOptions
```

Comment:
I'm really impressed and happy these are the only public API changes needed.

My memory is that this was some of the earliest changes made in the feature branch. Given all the knowledge gained since then, particularly all the brokered service work done to make codespaces scenarios work, are these public API changes still desired?  Knowing what we know now, are there better ways to handling changing package sources from VS options?

I'm ok with these changes, but I don't think these changes are beneficial to customers using `NuGet.Configuration` in their own apps, so if we can keep our public APIs a bit simpler (not add these new APIs), I think that would be better.

Reply:
we'd need to spend 1-2 days minimum to change the approach, but yes, we could. 
if we did that, i'd argue to do post preview 3.1 (current merge target) --- and it would delay our search work.
@aortiz-msft should be involved to decide.

Reply:
I'm fine with it being done after preview 3.1. But if it gets done after the last insertion into what becomes NuGet 5.8 GA, then the NuGet 5.8 SDK (nupkgs) will contain these APIs and removing them in NuGet 5.9 will become a breaking change.

So we need to decide either:
* these NuGet SDK changes are good enough and we'll support it "forever"
* or, commit to removing them before NuGet/VS goes GA, even if that means we do a QB insertion.

Will QB accept an insertion for NuGet with a commit that provides zero changes to VS experience (not even a bug fix), just because we want our nupkg/SDK to remove new APIs that we don't want to support?

Reply:
Yes, QB will accept insertions.
Will drive consensus with @dtivel @sbanni and team.
tracking via: https://github.com/nuget/home/issues/10015